### PR TITLE
Implement additional intersection methods

### PR DESCRIPTION
**Pull request by wrestledBearOnce**
Closes: #794

### CODE REVIEW

File: src/PointCloud/Core/PointCloudOctant.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿using Fusee.Math.Core;
+﻿using Fusee.Engine.Core.Primitives;
```
Comment (RedImp1470) [inline src/PointCloud/Core/PointCloudOctant.cs:1]: I don't think we should reference the `Primitives.Sphere` here (same goes for the comment in `InsideOrIntersectingSphere`) as we only need the mathematical construct "sphere" and not the renderable mesh.

File: src/PointCloud/Core/PointCloudOctant.cs [hunk 2]
```diff
@@ -193,6 +194,56 @@ public bool Intersects(double3 point)
             (point.z >= Min.z && point.z <= Max.z);
         }
 
+
+        /// <summary>
+        /// Returns true if the shape is completely inside or is intersecting this <see cref="PointCloudOctant"/>
+        /// </summary>
+        /// <param name="invSphereModel">The inverse model matrix which is currently used to render the <see cref="Sphere"/> (can e. g. be displayed as an ellipsoid)</param>
+        /// <returns></returns>
```
Comment (RedImp1470) [inline src/PointCloud/Core/PointCloudOctant.cs:202]: Maybe:

///Returns true if an ellipsoid is completely inside or is intersecting this <see cref="PointCloudOctant"/>.
///<param name="invSphereModel">The inverse transformation matrix that transforms an ellipsoid into a sphere.</param>

File: src/PointCloud/Core/PointCloudOctant.cs [hunk 3]
```diff
@@ -193,6 +194,56 @@ public bool Intersects(double3 point)
...
+        public bool InsideOrIntersectingSphere(double4x4 invSphereModel)
```
Comment (RedImp1470) [inline src/PointCloud/Core/PointCloudOctant.cs:203]: When I think about it now the method should probably be named `InsideOrIntersectingEllipsoid`. For `InsideOrIntersectingSphere` we only need the center and radius of the sphere and don't need the matrix multiplication that comes with the inverse transformation.

File: src/Math/Core/AABBd.cs [hunk 4]
```diff
@@ -254,6 +254,31 @@ public bool IntersectRay(RayD ray)
             return tmax >= M.Max(tmin, 0.0);
         }
 
+        /// <summary>
+        /// Returns the closest point to a point p, that lies on the surface of the <see cref="AABBd"/>.
+        /// </summary>
+        /// <param name="point">The reference point.</param>
+        /// <returns></returns>
+        public double3 ClosestPoint(double3 point)
```
Comment (RedImp1470) [inline src/Math/Core/AABBd.cs:262]: Needs tests. Add them or create an issue :)

File: src/Math/Core/AABBf.cs [hunk 5]
```diff
@@ -254,6 +254,32 @@ public bool IntersectRay(RayF ray)
             return tmax >= M.Max(tmin, 0.0);
         }
 
+        /// <summary>
+        /// Returns the closest point to a point p, that lies on the surface of the <see cref="AABBf"/>.
+        /// </summary>
+        /// <param name="point">The reference point.</param>
+        /// <returns></returns>
+        public float3 ClosestPoint(float3 point)
+        {
```
Comment (RedImp1470) [inline src/Math/Core/AABBf.cs:263]: Needs tests. Add them or create an issue :)

**Review by RedImp1470 — CHANGES_REQUESTED** [review]

File: src/Math/Core/AABBf.cs (hunk 5 above)
Comment (wrestledBearOnce) [reply src/Math/Core/AABBf.cs:263]: https://github.com/FUSEEProjectTeam/Fusee/issues/803

File: src/Math/Core/AABBd.cs (hunk 4 above)
Comment (wrestledBearOnce) [reply src/Math/Core/AABBd.cs:262]: https://github.com/FUSEEProjectTeam/Fusee/issues/803

**Review by RedImp1470 — APPROVED** [review]

**Pull request merged**